products/microsoft_defender_for_endpoints.py

```diff
@@ -154,10 +161,13 @@ def nested_process_search(self, tag: Tag, criteria: dict, base_query: dict) -> N
                     if isinstance(terms, list):
                         for query_entry in terms:
                             query_entry += f" {query_base}" if query_base != '' else ''
+                            if self._limit > 0: query_entry += f"| limit {str(self._limit)}"
```

Comment:
This line isn't needed since `self._limit` is applied after the query string is passed to `self.process_search`

```suggestion

```